server/api/groups.js

```diff
@@ -42,6 +42,7 @@ router.post('groups.list', auth(), pagination(), async ctx => {
       groupMemberships: groups
         .map(g => g.groupMemberships.slice(0, MAX_AVATAR_DISPLAY))
         .flat()
+        .filter(gM => gM.user)
```

Comment:
I think the filter needs to come before the slice — we want to display the first X valid group memberships

Reply:
You're not wrong, this was just easier 😛  – I'll change it.